src/freenet/node/CHKInsertSender.java

```diff
@@ -991,9 +991,9 @@ private boolean waitForBackgroundTransfers(BackgroundTransfer[] transfers) {
 					
 					if(logMINOR) Logger.minor(this, "Waiting: transfer completion=" + completedTransfers + " notification="+completedNotifications); 
 					try {
-						backgroundTransfers.wait(SECONDS.toMillis(100));
+						backgroundTransfers.wait(SECONDS.toMillis(10));
```

Comment:
This is not a sleep, but we're waiting for some other thread to wake us up (through `backgroundTransfers.notifyAll()`) after completing (successful or unsuccessful) a transfer, so we will loop until that has happened regardless of the maximum wait time here.

If anything, we should take our `deadline` into consideration that is checked at the start of the `while (true)` loop (currently set in the constructor: 1 minute for realtime, 5 minutes for bulk).